lib/pf/factory/condition/profile.pm

```diff
@@ -0,0 +1,124 @@
+package pf::factory::condition::profile;
+=head1 NAME
+
+pf::factory::condition::profile add documentation
+
+=cut
+
+=head1 DESCRIPTION
+
+pf::factory::condition::profile
+
+=cut
+
+use strict;
+use warnings;
+use Module::Pluggable search_path => 'pf::condition', sub_name => '_modules' , require => 1;
+our $DEFAULT_TYPE = 'ssid';
+our $PROFILE_FILTER_REGEX = qr/^(([^:]|::)+?):(.*)$/;
+use List::MoreUtils qw(any);
+
+our @MODULES;
+
+sub factory_for {'pf::condition'};
+
+my $DEFAULT_CONDITION = 'key';
+
+our %PROFILE_FILTER_TYPE_TO_CONDITION_TYPE = (
+    'network'         => {type => 'network',    key  => 'last_ip'},
+    'node_role'       => {type => 'equals',        key  => 'category'},
+    'connection_type' => {type => 'equals',        key  => 'last_connection_type'},
+    'port'            => {type => 'equals',        key  => 'last_port'},
+    'realm'           => {type => 'equals',        key  => 'realm'},
+    'ssid'            => {type => 'equals',        key  => 'last_ssid'},
+    'switch'          => {type => 'equals',        key  => 'last_switch'},
+    'switch_port'     => {type => 'key_couple', key1 => 'last_switch', key2 => 'last_port'},
+    'uri'             => {type => 'equals',        key  => 'last_uri'},
+    'vlan'            => {type => 'equals',        key  => 'last_vlan'},
+);
+
+sub modules {
+    my ($class) = @_;
+    unless(@MODULES) {
+        @MODULES = $class->_modules;
+    }
+    return @MODULES;
+}
+
+sub instantiate {
+    my ($class, @args) = @_;
+    my $condition;
+    my ($type,$data) = $class->getData(@args);
+    if ($data) {
+        if($type eq 'key_couple'){
+            my ($value1, $value2) = split(/-/, $data->{value});
+            my $cond1 = pf::condition::key->new(key => $data->{key1}, condition => pf::condition::equals->new(value => $value1));
+            my $cond2 = pf::condition::key->new(key => $data->{key2}, condition => pf::condition::equals->new(value => $value2));
+            my $cond_and = pf::condition::all->new(conditions => [$cond1, $cond2]);
+            return $cond_and;
+        }
+        else{
+            my $subclass = $class->getModuleName($type);
+            $condition = $subclass->new($data);
+            return pf::condition::key->new(key => $data->{key}, condition => $condition);
+        }
+    }
+}
+
+sub getModuleName {
+    my ($class, $type) = @_;
+    my $mainClass = $class->factory_for;
+    die "type is not defined" unless defined $type;
+    my $subclass = "${mainClass}::${type}";
+    die "$type is not a valid type of $mainClass" unless any {$_ eq $subclass} $class->modules;
+    $subclass;
+}
+
+sub getData {
+    my ($class, $filter) = @_;
+    my ($type, $value);
+    #Split parse the filter by type and value
+    if ($filter =~ $PROFILE_FILTER_REGEX ) {
+        $type  = $1;
+        $value = $3;
+    } else {
+        #If there is no type defined to support older filters (3.5.0)
```

Comment:
I think it should now be time to remove those. 

PacketFence 3.5 configuration is definitly deprecated